DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/mmhost/README.md

```diff
@@ -0,0 +1,372 @@
+| `awidth<width>`          | 41            | Width of the Avalon memory-mapped host address bus in bits. 
+| `dwidth<width>`          | 64            | Width of the Avalon memory-mapped host data bus in bits. 
+| `latency<value>`         | 1             | Read/Write latency for downstream fixed-latency Avalon memory-mapped agent. For a variable-latency Avalon memory-mapped agent, set `latency<0>`. 
+| `read_write_mode<mode>`  | `read_write`  | Port direction of the interface. (`read_write`, `read` or `write`) 
+| `maxburst<value>`        | 1             | Maximum number of data transfers that can associate with a read or write request. 
+| `alignment<alignment>`   | 1          | Alignment of the Avalon memory-mapped host interface
+
+These parameters can be used to improve the performance of `Example 1` by ensuring that each pointer points to data in a dedicated Avalon memory-mapped agent memory, like this:
+
+![](assets/dedicated_avhost.svg)
+
+#### Example 3: A kernel with dedicated Avalon memory-mapped host interfaces
+(Code can be found under `part3_hosts`).
+```c++
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+constexpr int kBL3 = 3;
+constexpr int kAlignment = 4;
+
+struct MultiMMIP {
+  // Each annotated pointer is configured with a unique `buffer_location`,
+  // resulting in three unique Avalon memory-mapped host interfaces.
+  using XProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL1>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_read});
+  using YProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL2>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_read});
+  using ZProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL3>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_write});
+
+  sycl::ext::oneapi::experimental::annotated_arg<int *, XProps> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, YProps> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ZProps> z;
+
+  int size;
+
+  void operator()() const {
+#pragma unroll 4
```

Comment:
@whitepau Is there a reason for unrolling by 4 here? The datawidth is 32 bits only.

Reply:
I think it's not needed here and can be removed.